crates/oxc_linter/src/config/config_builder.rs

```diff
@@ -134,6 +137,13 @@ impl ConfigStoreBuilder {
                                 builder = builder.with_rule(rule);
                             }
                             builder = builder.and_plugins(extended_config_store.plugins(), true);
+                            if !extended_config_store.overrides.is_empty() {
+                                let overrides =
+                                    std::mem::take(&mut extended_config_store.overrides);
+                                for r#override in overrides {
+                                    builder = builder.with_override(r#override);
```

Comment:
Can we add a test for `extends` and `files` combination which targets the same directory:
Example:

`.oxlintrc.json`
````json
{
  "rules": { "no-debugger": "error"  },
  "overrides": [
    {
      "files": ["config/*.ts", "config/*.js"],
      "rules": { "no-debugger": "off" }
    }
  ]
}
````

and `config/.oxclintrc.json`:

````json
{
  "extends": ["../.oxlintrc.json"],
  "rules": { "no-debugger": "warn" },
}
````

Reply:
Thanks, I've added this test case. I checked too and our behavior here is aligned with eslint too (this results in warn on files in `config`)